source/base/parameter_handler.cc

```diff
@@ -1710,46 +1731,61 @@ bool ParameterHandler::read_input (const std::string &filename,
       std::ifstream file_stream (openname.c_str());
       AssertThrow(file_stream, ExcIO());
 
-      return read_input (file_stream, filename, last_line);
+      parse_input (file_stream, filename, last_line);
     }
   catch (const PathSearch::ExcFileNotFound &)
     {
-      std::cerr << "ParameterHandler::read_input: could not open file <"
-                << filename << "> for reading." << std::endl;
       if (!optional)
         {
-          std:: cerr << "Trying to make file <"
-                     << filename << "> with default values for you." << std::endl;
           std::ofstream output (filename.c_str());
-          if (output)
-            print_parameters (output, (write_compact ? ShortText : Text));
+          print_parameters (output, (write_compact ? ShortText : Text));
         }
     }
-  return false;
+  return true;
```

Comment:
we are returning `true` unconditionally here because for all other error cases, an exception has been thrown?